kconfiglib.py

```diff
@@ -826,6 +826,69 @@ def _load_config(self, filename, replace):
                 if not choice._was_set:
                     choice.unset_value()
 
+    def write_autoconf(self, filename,
+                       header="/* Generated by Kconfiglib (https://github.com/ulfalizer/Kconfiglib) */\n"):
+        r"""
+        Writes out symbol values as a C header file.
+
+        filename:
+          Self-explanatory.
+
+        header (default: "/* Generated by Kconfiglib (https://github.com/ulfalizer/Kconfiglib) */\n"):
+          Text that will be inserted verbatim at the beginning of the file. You
+          would usually want it enclosed in "/* */" to make it a C comment,
+          and include a final terminating newline.
+        """
+        with open(filename, "w") as f:
+
+            # Small optimization
+            write = f.write
+            write(header)
+
+            # Avoid duplicates
+            for sym in self.defined_syms:
+                sym._written = False
+
+            node = self.top_node.list
+            if not node:
+                # Empty configuration
+                return
+
+            while 1:
+                item = node.item
+                if isinstance(item, Symbol):
+                    if not item._written:
+                        if item._write_to_conf:
+                            if item.orig_type in (BOOL, TRISTATE) and item.str_value != "n":
+                                suffix = "_MODULE" if item.str_value == "m" else ""
+                                write("#define {}{}{} 1\n"
+                                      .format(self.config_prefix, item.name,
+                                              suffix))
+                            elif item.orig_type == STRING:
+                                write('#define {}{} "{}"\n'
+                                      .format(self.config_prefix, item.name,
+                                              item.str_value))
+                            elif item.orig_type in (INT, HEX):
+                                write("#define {}{} {}\n"
+                                      .format(self.config_prefix, item.name,
+                                              item.str_value))
```

Comment:
Could do this, just to match the C implementation:

```python
if not val.startswith(("0x", "0X")):
    val = "0x" + val
```

Looks fine other than that!

I'm used to Gerrit. Think you might have to force push the branch or something to address review comments. :thinking:

Reply:
Yeah, I do it all the time with GitHub :)
Addressed but I couldn't keep it within the 80-char width limit, Python was throwing me errors.

Reply:
No worries.

If you want to split something that's not enclosed in brackets over multiple lines in Python, you need to add a \ at the end of the line.